src/rust-crypto/rust-crypto.ts

```diff
@@ -610,15 +610,17 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      * Implementation of {@link CryptoApi#isCrossSigningReady}
      */
     public async isCrossSigningReady(): Promise<boolean> {
-        const { publicKeysOnDevice, privateKeysInSecretStorage, privateKeysCachedLocally } =
-            await this.getCrossSigningStatus();
+        const { privateKeysInSecretStorage, privateKeysCachedLocally } = await this.getCrossSigningStatus();
         const hasKeysInCache =
             Boolean(privateKeysCachedLocally.masterKey) &&
             Boolean(privateKeysCachedLocally.selfSigningKey) &&
             Boolean(privateKeysCachedLocally.userSigningKey);
 
-        // The cross signing is ready if the public and private keys are available
-        return publicKeysOnDevice && (hasKeysInCache || privateKeysInSecretStorage);
+        const identity = await this.getOwnIdentity();
+
+        // The cross signing is ready if cross-signing is set up, the identity is trusted,
+        // and private keys are available somewhere.
+        return !!identity?.isVerified() && (hasKeysInCache || privateKeysInSecretStorage);
     }
```

Comment:
the behaviour of `isCrossSigningReady` is defined in `CryptoApi`.

It says:

> Checks whether cross signing:
> - is enabled on this account and **trusted by this device**
> - has private keys either cached locally or stored in secret storage

... so this sounds like fixing a *bug* in rust crypto, rather than just "fix inconsistency" ?

Reply:
Right, updating the title